src/extension-runners/chromium.js

```diff
@@ -151,12 +151,37 @@ export class ChromiumExtensionRunner {
     });
   }
 
-  wssBroadcast(data: Object) {
-    for (const client of this.wss.clients) {
-      if (client.readyState === WebSocket.OPEN) {
-        client.send(JSON.stringify(data));
+  async wssBroadcast(data: Object) {
+    return new Promise((resolve) => {
+      const clients = this.wss ? Array.from(this.wss.clients) : null;
+
+      if (!clients || clients.length === 0) {
+        resolve();
       }
-    }
+
+      const webExtReloadComplete = async (message) => {
+        const msg = JSON.parse(message);
+
+        if (msg.type === 'webExtReloadExtensionComplete') {
+          clients.forEach((client) =>
+            client.removeListener('message', webExtReloadComplete)
+          );
+          resolve();
+        }
+      };
+
+      clients.forEach((client) => {
+        if (client.readyState === WebSocket.OPEN) {
+          client.on('message', webExtReloadComplete);
+          client.on('close', () => {
+            client.removeListener('message', webExtReloadComplete);
+            clients.splice(clients.indexOf(client), 1);
```

Comment:
Let's use a `Set` instead of an `Array`, then you can use `clients.delete(client)` instead of having to use `splice` / `indexOf`.

Whenever you remove an element from the set, also check whether it may be necessary to check whether the set is empty, i.e. `if (clients.size === 0) { resolve(); }`.